core/src/main/scala/sttp/tapir/Validator.scala

```diff
@@ -170,6 +171,17 @@ object Validator extends ValidatorMagnoliaDerivation with ValidatorEnumMacro {
     }.toList
   }
 
+  case class Coproduct[T](ctx: SealedTrait[Validator, T]) extends Single[T] {
```

Comment:
I'm not sure if we want to use magnolia's class in our coproducts. Means that we won't  be able to create a `Coproduct` by hand. Maybe we should have our own abstraction, which could also be used in `statusMapping`?

Reply:
ah now I saw the description :)

maybe we can start with a simple `SealedTrait` abstraction exposing only the `dispatch` method & `subtypes` map? And add an issue to re-use it for `statusMapping`?

Reply:
Yes, you are absolutely right, we need to change it in order to give users an option to define it by them self.